grails-app/services/com/unifina/service/PermissionService.groovy

```diff
@@ -502,10 +490,15 @@ class PermissionService {
 				perm.operation == op
 			}.each { Permission perm ->
 				revoked.add(perm)
+				List<Permission> childPermissions = Permission.findAllByParent(perm)
+				revoked.addAll(childPermissions)
 				try {
 					log.info("performRevoke: Trying to delete permission $perm.id")
 					Permission.withNewTransaction {
 						perm.delete(flush: true)
+						for (Permission childPerm: childPermissions) {
```

Comment:
Child permissions need to be deleted first before deleting the parent permission. (It might not matter here in which way you write the code, as it happens in the same transaction. If this code works, then that must be the case.)

Reply:
Yes this code works since it's the same transaction.